obkvrpc/connection_pool.go

```diff
@@ -68,35 +103,38 @@ func NewConnectionPool(option *PoolOption) (*ConnectionPool, error) {
 		rwMutexes:   make([]sync.RWMutex, 0, option.connPoolMaxConnSize),
 	}
 
-	connectionOption := NewConnectionOption(pool.option.ip, pool.option.port, pool.option.connectTimeout, pool.option.loginTimeout,
-		pool.option.tenantName, pool.option.databaseName, pool.option.userName, pool.option.password)
+	if option.maxConnectionAge > 0 || option.enableSLBLoadBalance {
+		pool.connMgr = NewConnectionMgr(pool)
+	}
 
 	for i := 0; i < pool.option.connPoolMaxConnSize; i++ {
-
-		connection := NewConnection(connectionOption)
-
 		ctx, _ := context.WithTimeout(context.Background(), pool.option.connectTimeout)
-		err := connection.Connect(ctx)
+		connection, err := pool.CreateConnection(ctx)
 		if err != nil {
-			return nil, errors.WithMessage(err, "connection connect")
-		}
-
-		ctx, _ = context.WithTimeout(context.Background(), pool.option.loginTimeout)
-		err = connection.Login(ctx)
-		if err != nil {
-			return nil, errors.WithMessage(err, "connection login")
+			return nil, errors.WithMessage(err, "create connection")
 		}
 
 		pool.connections = append(pool.connections, connection)
 		pool.rwMutexes = append(pool.rwMutexes, sync.RWMutex{})
 
 	}
 
+	if pool.connMgr != nil {
+		pool.connMgr.start()
+	}
+
 	return pool, nil
 }
 
+// GetConnection Find an unexpired and active connection to use
+// In theory, all connection won't expire at the same time
```

Comment:
I guess we could print an error when all connections are invalid and return an error or nil.

Reply:
done